#! /usr/bin/env python

# Mutual camera localization configuration file

PACKAGE='mutual_camera_localization'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name             Type      Level Description             Default Min   Max
gen.add("gaussian_sigma", double_t,   0,    "The standard deviation of the Gaussian blurring filter", 0.1, 0, 6)
gen.add("min_blob_area", double_t,   0,    "Smallest blob area found", 100,     0,     300)
gen.add("max_blob_area", double_t,   0,    "Largest blob area found", 2000,     0,     4000)
gen.add("max_width_height_distortion", double_t, 0, "Max height width distortion of blob", 0.6, 0, 1)
gen.add("max_circular_distortion", double_t, 0, "Max deviation from roundness", 0.8, 0, 10)

gen.add("line_angle_tolerance", double_t, 1, "Tolerance on the line angle between the two line", 0.05, 0, 1)
gen.add("hor_line_angle", double_t, 1, "Max horizontal angle of the cube", 0.35, 0, 1)
gen.add("pos_ratio", double_t, 1, "Positional ratio of the three LED", 3, 0, 4)
gen.add("pos_ratio_tolerance", double_t, 1, "Tolerance on the positional ratio", 0.6, 0, 1)
gen.add("radius_ratio_tolerance", double_t, 1, "Tolerance on the radius ratio", 0.3, 0, 1)
gen.add("ratio_int_tolerance", double_t, 1, "Tolerance on the ratio intensity of the led", 0.25, 0, 1.5)
gen.add("ratio_ellipse_max", double_t, 1, "Max ellipse ratio", 1.5, 0, 3)
gen.add("ratio_ellipse_min", double_t, 1, "Min ellipse ratio", 0.65, 0, 3)

gen.add("pos_left_led_cam_a", double_t, 0, "Position of the led on the left side of the camera A(Always positive)", 0.13, 0, 2)
gen.add("pos_right_led_cam_a", double_t, 0, "Position of the led on the right side of the camera A(Always positive)", 0.14, 0, 2)
gen.add("pos_left_led_cam_b", double_t, 0, "Position of the led on the left side of the camera B(Always positive)", 0.11, 0, 2)
gen.add("pos_right_led_cam_b", double_t, 0, "Position of the led on the right side of the camera B(Always positive)", 0.125, 0, 2)


exit(gen.generate(PACKAGE, "mutual_camera_localization", "MutualCameraLocalization"))
